src/datautils/wellboreItemShapeGenerator.ts

```diff
@@ -57,56 +52,93 @@ export const cementDiameterChangeDepths = (
+  const uniqDepths = uniq(trimmedChangedDepths);
 
   return uniqDepths.sort((a: number, b: number) => a - b);
 };
 
-export const calculateCementDiameter =
-  (innerCasing: Casing[], nonAttachedCasings: Casing[], holes: HoleSize[]) =>
-  (
-    depth: number,
-  ): {
-    md: number;
-    innerDiameter: number;
-    outerDiameter: number;
-  } => {
-    const defaultCementWidth = 100; // Default to flow cement outside to show error in data
-
-    const innerCasingAtDepth = innerCasing.find((casing) => casing.start <= depth && casing.end >= depth);
-    const innerDiameter = innerCasingAtDepth ? innerCasingAtDepth.diameter : 0;
-
-    const outerCasings = nonAttachedCasings.filter((casing) => casing.innerDiameter > innerDiameter);
-    const holeAtDepth = holes.find((hole) => hole.start <= depth && hole.end >= depth && hole.diameter > innerDiameter);
-    const outerCasingAtDepth = outerCasings
-      .filter((d) => d)
-      .sort((a, b) => a.innerDiameter - b.innerDiameter) // ascending
-      .find((casing) => casing.start <= depth && casing.end >= depth && casing.diameter > innerDiameter);
-
-    let outerDiameter;
-    if (outerCasingAtDepth) {
-      outerDiameter = outerCasingAtDepth.innerDiameter;
-    } else if (holeAtDepth) {
-      outerDiameter = holeAtDepth.diameter - HOLE_OUTLINE;
-    } else {
-      outerDiameter = defaultCementWidth;
-    }
+export const findCementOuterDiameterAtDepth = (innerCasing: Casing[], nonAttachedCasings: Casing[], holes: HoleSize[], depth: number): number => {
+  const defaultCementWidth = 100; // Default to flow cement outside to show error in data
+
+  const innerCasingAtDepth = innerCasing.find((casing) => casing.start <= depth && casing.end >= depth);
+  const innerDiameter = innerCasingAtDepth ? innerCasingAtDepth.diameter : 0;
+
+  const outerCasings = nonAttachedCasings.filter((casing) => casing.innerDiameter > innerDiameter);
+  const holeAtDepth = holes.find((hole) => hole.start <= depth && hole.end >= depth && hole.diameter > innerDiameter);
+  const outerCasingAtDepth = outerCasings
+    .filter((d) => d)
+    .sort((a, b) => a.innerDiameter - b.innerDiameter) // ascending
+    .find((casing) => casing.start <= depth && casing.end >= depth && casing.diameter > innerDiameter);
+
+  let outerDiameter;
+  if (outerCasingAtDepth) {
+    outerDiameter = outerCasingAtDepth.innerDiameter;
+  } else if (holeAtDepth) {
+    outerDiameter = holeAtDepth.diameter - HOLE_OUTLINE;
+  } else {
+    outerDiameter = defaultCementWidth;
+  }
+
+  return outerDiameter;
+};
+
+export const createComplexRopeSegmentsForCement = (
+  cement: Cement,
+  casings: Casing[],
+  holes: HoleSize[],
+  exaggerationFactor: number,
+  getPoints: (start: number, end: number, interestPoints: number[]) => MDPoint[],
+): ComplexRopeSegment[] => {
+  // Merge deprecated casingId and casingIds array
+  // TODO remove casingId now?
+  const casingIds = [cement.casingId, ...(cement.casingIds || [])].filter((id) => id);
```

Comment:
Let's remove the deprecated `casingId` 🥳

Reply:
This would depend on updating data in esv-intersection-data repository. Adding it to my TODO list for later when we get access to that repository.